modules/gapi/misc/python/test/test_gapi_sample_pipelines.py

```diff
@@ -522,5 +375,288 @@ def test_custom_op_goodFeaturesToTrack(self):
+                return GSumImpl.last_result
+
+
+        g_in  = cv.GArray.Int()
+        comp  = cv.GComputation(cv.GIn(g_in), cv.GOut(GSum.on(g_in)))
+
+        s = comp.apply(cv.gin([1, 2, 3, 4]), args=cv.compile_args(cv.gapi.kernels(GSumImpl)))
+        self.assertEqual(10, s)
+
+        s = comp.apply(cv.gin([1, 2, 8, 7]), args=cv.compile_args(cv.gapi.kernels(GSumImpl)))
+        self.assertEqual(18, s)
+
+        self.assertEqual(18, GSumImpl.last_result)
+
+
+    def test_opaq_with_custom_type(self):
+        @cv.gapi.op('custom.op', in_types=[cv.GOpaque.Any, cv.GOpaque.String], out_types=[cv.GOpaque.Any])
+        class GLookUp:
+            @staticmethod
+            def outMeta(opaq_desc0, opaq_desc1):
+                return cv.empty_gopaque_desc()
+
+        @cv.gapi.kernel(GLookUp)
+        class GLookUpImpl:
+            @staticmethod
+            def run(table, key):
+                return table[key]
+
+
+        g_table = cv.GOpaque.Any()
+        g_key   = cv.GOpaque.String()
+        g_out   = GLookUp.on(g_table, g_key)
+
+        comp = cv.GComputation(cv.GIn(g_table, g_key), cv.GOut(g_out))
+
+        table = {
+                    'int':   42,
+                    'str':   'hello, world!',
+                    'tuple': (42, 42)
+                }
+
+        out = comp.apply(cv.gin(table, 'int'), args=cv.compile_args(cv.gapi.kernels(GLookUpImpl)))
+        self.assertEqual(42, out)
+
+        out = comp.apply(cv.gin(table, 'str'), args=cv.compile_args(cv.gapi.kernels(GLookUpImpl)))
+        self.assertEqual('hello, world!', out)
+
+        out = comp.apply(cv.gin(table, 'tuple'), args=cv.compile_args(cv.gapi.kernels(GLookUpImpl)))
+        self.assertEqual((42, 42), out)
+
+
+    def test_array_with_custom_type(self):
+        @cv.gapi.op('custom.op', in_types=[cv.GArray.Any, cv.GArray.Any], out_types=[cv.GArray.Any])
+        class GConcat:
+            @staticmethod
+            def outMeta(arr_desc0, arr_desc1):
+                return cv.empty_array_desc()
+
+        @cv.gapi.kernel(GConcat)
+        class GConcatImpl:
+            @staticmethod
+            def run(arr0, arr1):
+                return arr0 + arr1
+
+        g_arr0 = cv.GArray.Any()
+        g_arr1 = cv.GArray.Any()
+        g_out  = GConcat.on(g_arr0, g_arr1)
+
+        comp = cv.GComputation(cv.GIn(g_arr0, g_arr1), cv.GOut(g_out))
+
+        arr0 = [(2, 2), 2.0]
+        arr1 = [3,    'str']
+
+        out = comp.apply(cv.gin(arr0, arr1),
+                         args=cv.compile_args(cv.gapi.kernels(GConcatImpl)))
+
+        self.assertEqual(arr0 + arr1, out)
+
+
+    def test_raise_in_kernel(self):
+        @cv.gapi.op('custom.op', in_types=[cv.GMat, cv.GMat], out_types=[cv.GMat])
+        class GAdd:
+            @staticmethod
+            def outMeta(desc0, desc1):
+                return desc0
+
+        @cv.gapi.kernel(GAdd)
+        class GAddImpl:
+            @staticmethod
+            def run(img0, img1):
+                raise 'Error'
```

Comment:
This is not correct:
```
    raise 'Error'
TypeError: exceptions must derive from BaseException
```

Reply:
What do you mean ? Just derive it from `BaseException` ?

Reply:
This is a Python error message.
Use proper classes for raising exceptions in Python, e.g. `Exception`

Reply:
Done